### PR TITLE
Add go.mod sync workflow to sync Meshery and Meshery-extensions edge

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: Ashish Tiwari <ashishjaitiwari15112000@gmail.com>

**Notes for Reviewers**

This PR adds a workflow which:
1. Gets triggered only when the go.mod on meshery:master changes
2. It results in go.mod sync utility to run on meshery-extensions repo and commit the synced go.mod

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Oooh, nice: a PR label and description. No use of `[<component>]` in the title, though, so not immediately clear what component this primarily affects.

**Comment by leecalcote** [comment]
Oh, I see that the `area/ci` label was a freebie, automatically added.

File: .github/workflows/go-mod-sync.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+on:
+  push:
+    branches:
+      - "master"
+    paths:
+      - "go.mod"
+jobs:
+  sync:
+    name: sync
+    runs-on: ubuntu-18.04
```
Comment (leecalcote) [inline .github/workflows/go-mod-sync.yaml:11]: @StarTrooper08 @acald-creator we've move to 22, correct?

File: .github/workflows/go-mod-sync.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Meshery extensions go.mod sync
```
Comment (leecalcote) [inline .github/workflows/go-mod-sync.yaml:1]: @Revolyssup, perhaps, this workflow should be just another step in the existing Edge and Stable release workflows?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: .github/workflows/go-mod-sync.yaml (hunk 1 above)
Comment (StarTrooper08) [reply .github/workflows/go-mod-sync.yaml:11]: Yes, we had move GitHub actions workflow Ubuntu version specifically to 22.04.
Cause it gives some warnings if we are using 18.04 version or specify latest.

File: .github/workflows/go-mod-sync.yaml (hunk 2 above)
Comment (Revolyssup) [reply .github/workflows/go-mod-sync.yaml:1]: This only gets triggered when the go.mod changes. The rest are general.

Comment (leecalcote) [reply .github/workflows/go-mod-sync.yaml:1]: OK. 👍

File: .github/workflows/go-mod-sync.yaml (hunk 1 above)
Comment (leecalcote) [reply .github/workflows/go-mod-sync.yaml]: Thanks for confirming, @StarTrooper08 :thu

**Review by leecalcote — APPROVED** [review]

File: .github/workflows/go-mod-sync.yaml (hunk 1 above)
Comment (StarTrooper08) [reply .github/workflows/go-mod-sync.yaml]: Happy to Help😄

**Pull request merged**